### DISCUSSION TITLE
协程池里面的worker是并发执行的吗

### DISCUSSION THREAD

**Discussion by lizzyilike** [panjf2000/ants#266] [Q&A] [answered]
协程池里面的worker是并发执行的吗，例如将一批任务用不同的协程池大小执行，总耗时是否有变化

**Comment by panjf2000** [comment, marked as answer]
是并发执行的，至于你的问题，不同大小的 pool 执行同一批任务，是有可能耗时不一样的，取决于你的任务有多少，如果超过了 pool 的容量，则需要等待其他任务执行完，这样就会花更多时间。

**Reply by lizzyilike** [reply]
多谢回答，我的问题知道了，我这边有将结果放到channel，然后异步处理的逻辑，由于处理结果是串行处理的，处理的速度赶不上生产的速度，导致生产的时候往chanel发消息总是阻塞

**Reply by ljx0520** [reply]
请问你是怎么解决这个问题的？

**Reply by lizzyilike** [reply]
> 请问你是怎么解决这个问题的？

再用多个协程去消费

**Reply by josephlabeleven** [reply]
我现在就是一边在这写入一边在另外一头读取

**Reply by lizzyilike** [reply]
或者你收到结果后，要是处理结果不复杂的话，没有必要把结果放到channel，直接处理就行

**Reply by josephlabeleven** [reply]
嗯嗯好的， 我是想收集结果之后再batch insert

**Reply by lizzyilike** [reply]
> 我现在就是一边在这写入一边在另外一头读取

是的，我也是这么搞的，读取的时候也是用多个协程并发去读取channel里面的内容

**Reply by lizzyilike** [reply]
> 嗯嗯好的， 我是想收集结果之后再batch insert

不过我还是建议处理结果用channel接收，再消费channel的模式，这样便于解耦，前提是你要并发消费

**Reply by josephlabeleven** [reply]
嗯嗯好的 感谢
